scripts/generate-deploy-yamls.sh

```diff
@@ -1,52 +1,158 @@
+
+# Internal variables tunable by options
+cores=
+moac_debug="-v"
+output_dir="$SCRIPTDIR/../deploy"
+pools=
+profile=
+pull_policy=
+registry=
+tag=
+
+help() {
+  cat <<EOF
+
+Usage: $0 [OPTIONS] <PROFILE>
 
-Generate (some) deployment YAMLs from the helm chart and store them to deploy/
-in the repo. If -t is specified do not put them to deploy/ but rather to the
-directory given.
+Common options:
+  -c <cores>       # of cpu cores for mayastor overriding the profile's default.
+  -h/--help        Display help message and exit.
+  -o <output_dir>  Directory to store the generated yaml files (default $output_dir)
+  -p <node,device> Node name and associated pool device (the option may repeat).
+  -r <registry>    Docker image registry of mayastor images (default none).
+  -t <tag>         Tag of mayastor images overriding the profile's default.
+
+Profiles:
+  develop:   Used by developers of mayastor.
+  release:   Recommended for stable releases deployed by users.
+  test:      Used by mayastor e2e tests.
 EOF
-	exit 1
-fi
+}
 
-SCRIPTDIR="$(realpath "$(dirname "$0")")"
+# Parse common options
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    -c)
+      shift
+      cores=$1
+      ;;
+    -h|--help)
+      help
+			exit 0
+      ;;
+    -o)
+      shift
+      output_dir=$1
+      ;;
+    -p)
+      shift
+      pools="$pools $1"
+      ;;
+    -r)
+      shift
+      registry=$1
+      ;;
+    -t)
+      shift
+      tag=$1
+      ;;
+    -*)
+      echo "Unknown option: $1"
+      help
+      exit 1
+      ;;
+    *)
+      profile=$1
+			shift
+			break
+      ;;
+  esac
+  shift
+done
 
-if [ "$1" = "-t" ]; then
-	TARGET_DIR="$2"
-	shift 2
-else
-	TARGET_DIR="$SCRIPTDIR/../deploy"
-fi
-if [ ! -d "$TARGET_DIR" ]; then
-	mkdir -p "$TARGET_DIR"
+# The space after profile name is reserved for profile specific options which
+# we don't have yet.
+if [ "$#" -gt 0 ]; then
+	help
+	exit 1
 fi
 
-if [ "x$2" = x ]; then
-	mayastor_images_repo="NONE"
+# In most of the cases the tag will be a specific version that does not change
+# so save dockerhub bandwidth and don't always pull the image.
+if [ -n "$tag" ]; then
+	pull_policy=IfNotPresent
 else
-	mayastor_images_repo="$2"
+	pull_policy=Always
 fi
 
+# Set profile defaults
+case "$profile" in
+  "develop")
+		[ -z "$cores" ] && cores=1
+		[ -z "$tag" ] && tag=develop
+		moac_debug="-vv"
+	  ;;
+  "release")
+		[ -z "$cores" ] && cores=1
+		[ -z "$tag" ] && tag=latest
+	  ;;
+  "test")
+		[ -z "$cores" ] && cores=2
+		[ -z "$tag" ] && tag=ci
+	  ;;
+  *)
+		echo "Missing or invalid profile name. Type \"$0 --help\""
+		exit 1
+	  ;;
+esac
+
 set -u
 
-if ! which helm > /dev/null 2>&1; then
-	echo "Install helm to path >v3.4.1"
-	echo "https://github.com/helm/helm/releases/tag/v3.4.1"
+if ! which helm >/dev/null 2>&1; then
+	echo "Install helm (>v3.4.1) to PATH"
 	exit 1
 fi
 
+if [ ! -d "$output_dir" ]; then
+	mkdir -p "$output_dir"
+fi
+
 tmpd=$(mktemp -d /tmp/generate-deploy.sh.XXXXXXXX)
```

Comment:
```suggestion
tmpd=$(mktemp -d /tmp/generate-deploy-yamls.sh.XXXXXXXX)
```

please fix my inconsistency

Reply:
will fix 👍